pkg/spire/config/config.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// SpireConfig holds the images reference for a number of container images used
+// across tektoncd pipelines.
+type SpireConfig struct {
+	// The trust domain corresponds to the trust root of a SPIFFE identity provider.
+	TrustDomain string
+	// Path to the spire agent socket defined by the CSI driver
+	SocketPath string
+	// Spire server address
+	ServerAddr string
+	// Prefix to attach to the node name when registering it with the spire server
+	NodeAliasPrefix string
+
+	// MockSpire only to be used for testing the controller, will not exhibit
+	// all characteristics of spire since it is only being used in the context
+	// of process memory.
+	MockSpire bool
+}
+
+// Validate returns an error if any image is not set.
+func (c SpireConfig) Validate() error {
+	var unset []string
+	for _, f := range []struct {
+		v, name string
+	}{
+		{c.TrustDomain, "spire-trust-domain"},
+		{c.SocketPath, "spire-socket-path"},
+		{c.ServerAddr, "spire-server-addr"},
+		{c.NodeAliasPrefix, "spire-node-alias-prefix"},
+	} {
+		if f.v == "" {
+			unset = append(unset, f.name)
+		}
+	}
+	if len(unset) > 0 {
+		sort.Strings(unset)
+		return fmt.Errorf("found unset image flags: %s", unset)
```

Comment:
I don't think the error is correct here (no image involevd)

Reply:
I think it will be great to make this error more descriptive. This error will be met when `SpireConfig` is missing all four required flags which in turn means no image initialized 🤔

Reply:
changed to "found unset spire configuration flags"